### PR TITLE
Respond to SIGINT during plugin install

### LINKED ISSUE #9629: Pulumi does not respond to SIGNIT while command is ran from a child process
### What happened?

Pulumi does not respond to SIGNIT while command is ran from a child process. For instance, running `pulumi up` to force a creation of a venv (python program) results in a Pulumi program that does not respond to SIGINT. 

### Steps to reproduce

1. create a simple Pulumi python program. Eg- `pulumi new aws python`
2. delete the virtual environment. Eg - `rm -rf venv`.
3. `pulumi up`
4. SIGINT - CTRL + C during the update, while the virtual environment is being created.

### Expected Behavior

Pulumi should respond to SIGINT - CTRL + C and halt the program.

### Actual Behavior

Pulumi does not respond to the SIGINT - CTRL + C

### Versions used

CLI - 3.30.0

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by iwahbe**
# Description

This PR changes a `python.Command` to call `exec.CommandContext` and threads through the necessary `plugin.Context` and `context.Context`. While it touches many files, most files only contain passing through a instance of `context.Context`. 

Fixes #9629

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/cmd/pulumi/about_test.go [hunk 1]
```diff
@@ -46,7 +47,7 @@ func TestProjectRuntime(t *testing.T) {
 	version := strings.TrimSpace(strings.TrimPrefix(string(out), "Python "))
 
 	var runtime projectRuntimeAbout
-	runtime, err = getProjectRuntimeAbout(&workspace.Project{
+	runtime, err = getProjectRuntimeAbout(context.TODO(), &workspace.Project{
```
Comment (Frassle) [inline pkg/cmd/pulumi/about_test.go:50]: We seem inconsistent about Background or TODO in tests?

File: sdk/go/common/resource/plugin/context.go [hunk 2]
```diff
@@ -96,3 +110,18 @@ func (ctx *Context) Close() error {
 	}
 	return nil
 }
+
+// WithCancelChannel registers a close channel which will close the Context when the
```
Comment (Frassle) [inline sdk/go/common/resource/plugin/context.go:114]:
```suggestion
// WithCancelChannel registers a close channel which will close the returned Context when the
```

File: sdk/go/common/workspace/plugins.go [hunk 3]
```diff
@@ -720,6 +721,11 @@ func (info PluginInfo) installLock() (unlock func(), err error) {
 	}, nil
 }
 
+// Install installs a plugin's tarball into the cache. See InstallWithContext for details.
+func (info PluginInfo) Install(tgz io.ReadCloser, reinstall bool) error {
```
Comment (Frassle) [inline sdk/go/common/workspace/plugins.go:725]: It is very lame that we class this stuff as public api in sdk/go. I can't imagine anything but the engine cares about it.

File: sdk/go/common/workspace/plugins.go [hunk 4]
```diff
@@ -824,7 +830,8 @@ func (info PluginInfo) Install(tgz io.ReadCloser, reinstall bool) error {
 				return errors.Wrap(err, "installing plugin dependencies")
 			}
 		case "python":
-			if err := python.InstallDependencies(finalDir, "venv", false /*showOutput*/); err != nil {
+			// TODO: pass a context here
```
Comment (Frassle) [inline sdk/go/common/workspace/plugins.go:833]: You are passing a context here? Why the TODO? Was this meant for the nodejs case above?

Comment (iwahbe) [reply sdk/go/common/workspace/plugins.go]: I'm not 100% sure, but I should do that. Sorry about that.

**Review by Frassle — APPROVED** [review]

**Pull request merged**